### PR TITLE
[IMP] shopinvader: modify method copy for res.users obj

### PR DESCRIPTION
**Pull request by i-vyshnevska**
forward port https://github.com/shopinvader/odoo-shopinvader/pull/1115

### CODE REVIEW

**Comment by i-vyshnevska** [comment]
@simahawk @JordiMForgeFlow can you give a look at this PR

File: shopinvader_customer_validate/models/res_partner.py [hunk 1]
```diff
@@ -3,7 +3,8 @@
 # License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
```
Comment (simahawk) [inline shopinvader_customer_validate/models/res_partner.py]: this has to stay in core module

Comment (simahawk) [reply shopinvader_customer_validate/models/res_partner.py]: `has_shopinvader_user_active` can be moved to `shopinvader`

**Review by simahawk — CHANGES_REQUESTED** [review]

**Comment by i-vyshnevska** [comment]
@simahawk updated

File: shopinvader/models/res_partner.py [hunk 2]
```diff
@@ -43,6 +43,9 @@ class ResPartner(models.Model):
         compute_sudo=True,
         store=True,
     )
+    has_shopinvader_user_active = fields.Boolean(
```
Comment (simahawk) [inline shopinvader/models/res_partner.py]: this field should be computed as well otherwise it won't have any value and your query won't be useful at all

**Review by simahawk — CHANGES_REQUESTED** [review]

File: shopinvader/models/res_partner.py (hunk 2 above)
Comment (i-vyshnevska) [reply shopinvader/models/res_partner.py]: here I just defined it and in shopinvader_customer_validate it is overridden with compute method, or we should have a hook method here?

Comment (simahawk) [reply shopinvader/models/res_partner.py]: but is useless in you SQL if it's not populated... I don't get your point, sorry :)

Comment (i-vyshnevska) [reply shopinvader/models/res_partner.py]: the compute method is present and stayed in shopinvader_customer_validate
https://github.com/shopinvader/odoo-shopinvader/blob/220d9467eb13af5ea2fe37664e13dc8af65bfc30/shopinvader_customer_validate/models/res_partner.py#L18

Comment (simahawk) [reply shopinvader/models/res_partner.py]: ok, let's put it this way: what's the role of `has_shopinvader_user_active` here when you don't have `s_c_validate` installed?

Comment (i-vyshnevska) [reply shopinvader/models/res_partner.py]: none, if we don't have s_c_validate, has_shopinvader_user_active will have false value and behavior will be the same as before
I don't want to bring compute method in shopinvader as then I will need to bring more code regards partner states here
maybe a more graceful option is to add an option via a hook in constrain statement  and in s_c_validate return the proper condition

**Review by TDu — APPROVED** [review]

File: shopinvader/models/res_partner.py [hunk 3]
```diff
@@ -55,6 +55,37 @@ def _is_partner_duplicate_prevented(self):
...
+            FROM (
+                SELECT
+                    rp.id,
+                    rp.email,
+                    ROW_NUMBER() OVER (PARTITION BY rp.email) AS Row
+                FROM
+                    res_partner rp
+                WHERE rp.email is not null
+                    and rp.active = True
+                    and EXISTS (SELECT FROM shopinvader_partner sp WHERE sp.record_id = rp.id)
```
Comment (simahawk) [inline shopinvader/models/res_partner.py:76]: what about `AND sp.active`?

File: shopinvader_customer_validate/tests/test_partner_constrains.py [hunk 4]
```diff
@@ -0,0 +1,78 @@
+# Copyright 2018 ACSONE SA/NV
```
Comment (simahawk) [inline shopinvader_customer_validate/tests/test_partner_constrains.py]: bad copy paste?

**Review by simahawk — CHANGES_REQUESTED** [review]

**Comment by simahawk** [comment]
pls do not include the odoo version in commit msgs :pray: (talking about your commits, not the cherry-picked one)

File: shopinvader/models/res_partner.py (hunk 3 above)
Comment (i-vyshnevska) [reply shopinvader/models/res_partner.py:76]: shopinvader_partner doesn't have the active field, we have is_shopinvader_active but it is made computable so cannot be used in sql

**Comment by sebastienbeau** [comment]
@lmignon @Cedric-Pigeon I think this change the initial behaviour that you have implemented ? (The intial idea was to restrict on res_partner right independently of the backend?)

**Review by lmignon — APPROVED** [review]
LGTM. BTW, this constraint is here just because we enforced the email as shared identifier between a locomotive account and a shopinvader.partner into odoo. Since it's the responsibility of the authentication logic to provide the partner linked to a request, this constraint could be added only if it's required by the authentication. Moreover, the same problem will be back in such a case since the shopinvader_partner model will be removed in the new services implementation.

**Pull request merged**